### PR TITLE
Remove unnecessary controllers

### PR DESCRIPTION
**Pull request by CoderKevinZhang**
**What type of PR is this?**
> /kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kube-controller-manager/app/BUILD [hunk 1]
```diff
@@ -133,21 +89,15 @@ go_library(
         "//staging/src/k8s.io/client-go/kubernetes:go_default_library",
         "//staging/src/k8s.io/client-go/rest:go_default_library",
         "//staging/src/k8s.io/client-go/restmapper:go_default_library",
-        "//staging/src/k8s.io/client-go/scale:go_default_library",
```
Comment (cathyhongzhang) [inline cmd/kube-controller-manager/app/BUILD:136]: Do you need to remove this scale related library? Could you double check that it is not needed for controller scaling?

File: cmd/kube-controller-manager/app/BUILD [hunk 2]
```diff
@@ -40,71 +35,32 @@ go_library(
...
-        "//pkg/volume/glusterfs:go_default_library",
-        "//pkg/volume/hostpath:go_default_library",
         "//pkg/volume/iscsi:go_default_library",
-        "//pkg/volume/local:go_default_library",
-        "//pkg/volume/nfs:go_default_library",
         "//pkg/volume/portworx:go_default_library",
-        "//pkg/volume/quobyte:go_default_library",
         "//pkg/volume/rbd:go_default_library",
         "//pkg/volume/scaleio:go_default_library",
         "//pkg/volume/storageos:go_default_library",
```
Comment (cathyhongzhang) [inline cmd/kube-controller-manager/app/BUILD:66]: It seems that you remove some volume related library but keep some. Could you clarify why?

File: cmd/kube-controller-manager/app/BUILD [hunk 3]
```diff
@@ -4,17 +4,12 @@ go_library(
...
-        "bootstrap.go",
         "certificates.go",
         "cloudproviders.go",
         "controllermanager.go",
         "core.go",
         "import_known_versions.go",
         "plugins.go",
-        "policy.go",
-        "rbac.go",
     ],
```
Comment (cathyhongzhang) [inline cmd/kube-controller-manager/app/BUILD:13]: Why do you need to remove the policy.go and rbac.go? rbac.go seems to be related to role based access control.

Comment (CoderKevinZhang) [reply cmd/kube-controller-manager/app/BUILD:13]: policy.go is using disruption controller and rbac.go is using clusterroleaggregation controller. Neither of these two controllers is needed.

File: cmd/kube-controller-manager/app/BUILD (hunk 2 above)
Comment (CoderKevinZhang) [reply cmd/kube-controller-manager/app/BUILD:66]: these volume-related libraries are associated with persistentvolume-binder/attachdetach/persistentvolume-expander controllers

File: cmd/kube-controller-manager/app/BUILD (hunk 1 above)
Comment (CoderKevinZhang) [reply cmd/kube-controller-manager/app/BUILD:136]: the reason I remove client-go/scale is horizontalpodautoscaling controller is using it, and horizontalpodautoscaling controller is not needed

Comment (CoderKevinZhang) [reply cmd/kube-controller-manager/app/BUILD:136]: Memo - we need to implement scaling in the future

**Review by cathyhongzhang — APPROVED** [review]

**Review by cathyhongzhang — CHANGES_REQUESTED** [review]
Need to add back the endpoint controller. I am sorry for asking you to make this change. Since leader election for HA in Kubernetes can use endpoint object, we'd better add back the endpoint controller.

File: test/e2e/network/BUILD [hunk 4]
```diff
@@ -32,7 +32,6 @@ go_library(
     importpath = "k8s.io/kubernetes/test/e2e/network",
     deps = [
         "//pkg/apis/core:go_default_library",
-        "//pkg/controller/endpoint:go_default_library",
         "//pkg/master/ports:go_default_library",
```
Comment (cathyhongzhang) [inline test/e2e/network/BUILD:35]: why is this removed? it seems that it does not need to be removed.

File: test/e2e/network/service.go [hunk 5]
```diff
@@ -37,7 +37,6 @@ import (
 	"k8s.io/apimachinery/pkg/util/wait"
 	clientset "k8s.io/client-go/kubernetes"
 	cloudprovider "k8s.io/cloud-provider"
-	"k8s.io/kubernetes/pkg/controller/endpoint"
 	"k8s.io/kubernetes/test/e2e/framework"
```
[the remark this answers is not in the document]
Comment (CoderKevinZhang) [reply test/e2e/network/service.go:40]: I will add it back

File: test/e2e/network/BUILD (hunk 4 above)
Comment (CoderKevinZhang) [reply test/e2e/network/BUILD:35]: I will add it back

**Comment by cathyhongzhang** [comment]
LGTM

**Review by cathyhongzhang — APPROVED** [review]

**Pull request merged**